File: javascript/RemoveDuplicatedMessage.js

const input = [
    {
        id: "1",
        content: "We have to make an impact"
    },
    {
        id: "2",
        content: "Bro i need a jobbbb"
    },
    {
        id:"1", 
        content: "Million Naira deals ooo"
    }
]
let tracker = []
function normalizeMessages(messages) {
    if(!Array.isArray(messages)){
     return console.log("Invalid Data type")
    }
    messages.forEach(item => {
        let id = item.id 
        tracker.push(id)
        if(tracker.includes(id)){
           messages.pop(item)
        }
        
    })
   console.log(messages)
}

normalizeMessages(input)

File: python/RemoveDuplicatedMessage.py
input = [{"id":"1", "content":"Testing one"}, {"id":"2", "content":"Testing 2"}, {"id":"1", "content":"Testing one"}]     

output = []
tracker = []
def removeDuplicatedMessage(arr):
    if type(arr) != list:
       return  print("Invalid data type")
    for message in arr:
        if message["id"] in tracker:
            pass
        else:
            output.append(message)
            tracker.append(message["id"])
    arr = output    
    return arr
        
print(removeDuplicatedMessage(input)) 